### PR TITLE
r/aws_dax_cluster: Support encryption at rest

### PR DESCRIPTION
**Pull request by ewbankkit**
Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/5500.
Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSDAXCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSDAXCluster_basic -timeout 120m
=== RUN   TestAccAWSDAXCluster_basic
--- PASS: TestAccAWSDAXCluster_basic (669.50s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	669.513s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSDAXCluster_encryption'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSDAXCluster_encryption -timeout 120m
=== RUN   TestAccAWSDAXCluster_encryption
--- PASS: TestAccAWSDAXCluster_encryption (596.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	596.203s
```

### CODE REVIEW

File: aws/resource_aws_dax_cluster_test.go [hunk 1]
```diff
@@ -239,9 +275,9 @@ EOF
 func testAccAWSDAXClusterConfig(rString string) string {
 	return fmt.Sprintf(`%s
 		resource "aws_dax_cluster" "test" {
- 		  cluster_name       = "tf-%s"
+		  cluster_name       = "tf-%s"
 		  iam_role_arn       = "${aws_iam_role.test.arn}"
-		  node_type          = "dax.r3.large"
+		  node_type          = "dax.t2.small"
```
Comment (ewbankkit) [inline aws/resource_aws_dax_cluster_test.go]: Node type changed as encryption at rest is not supported for dax.r3.* node types:
* https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DAXEncryptionAtRest.html

File: aws/resource_aws_dax_parameter_group.go [hunk 2]
```diff
@@ -158,34 +158,3 @@ func resourceAwsDaxParameterGroupDelete(d *schema.ResourceData, meta interface{}
 
 	return nil
 }
-
```
Comment (ewbankkit) [inline aws/resource_aws_dax_parameter_group.go]: Moved these 2 functions to `structure.go`.

File: aws/resource_aws_dax_cluster_test.go [hunk 3]
```diff
@@ -137,6 +151,28 @@ func TestAccAWSDAXCluster_resize(t *testing.T) {
 					testAccCheckAWSDAXClusterExists("aws_dax_cluster.test", &dc),
 					resource.TestCheckResourceAttr(
 						"aws_dax_cluster.test", "replication_factor", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_dax_cluster.test", "server_side_encryption.0.enabled", "true"),
```
Comment (bflad) [inline aws/resource_aws_dax_cluster_test.go]: It looks like this check was extraneously added 😉 

```
--- FAIL: TestAccAWSDAXCluster_resize (1252.75s)
	testing.go:527: Step 2 error: Check failed: Check 3/3 error: aws_dax_cluster.test: Attribute 'server_side_encryption.0.enabled' expected "true", got "false"
```

File: aws/resource_aws_dax_cluster_test.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_dax_cluster_test.go]: 👍

File: aws/resource_aws_dax_cluster.go [hunk 4]
```diff
@@ -95,6 +96,21 @@ func resourceAwsDaxCluster() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      schema.HashString,
 			},
+			"server_side_encryption": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
```
Comment (bflad) [inline aws/resource_aws_dax_cluster.go]: I was able to remove the following by changing the schema definition:
* `Computed: true`
* `CustomizeDiff`
* `isDaxClusterOptionDisabled`

Check this out:

```go
"server_side_encryption": {
	Type:     schema.TypeList,
	Optional: true,
	MaxItems: 1,
	DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
		if old == "1" && new == "0" {
			return true
		}
		return false
	},
	Elem: &schema.Resource{
		Schema: map[string]*schema.Schema{
			"enabled": {
				Type:     schema.TypeBool,
				Optional: true,
				Default:  false,
			},
		},
	},
},
```

At some point I need to make that `DiffSuppressFunc` into its own function 😅 , its basically just there to ignore the container attribute being unset in the Terraform configuration (all child attribute differences still apply).

Comment (ewbankkit) [reply aws/resource_aws_dax_cluster.go]: 👍 I reused the logic from [`aws_dynamodb_table`](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_dynamodb_table.go#L43-L51) so we may want to refactor there also.

Comment (ewbankkit) [reply aws/resource_aws_dax_cluster.go]: Especially as it looks like with [AWS SDK v1.15.16](https://github.com/aws/aws-sdk-go/releases/tag/v1.15.16) `aws_dynamodb_table` SSE can be updated without recreating the table.

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**